UmpleToJava/UmpleTLTemplates/state_machine_StateDependentMethods_All.ump

```diff
@@ -0,0 +1,143 @@
+class UmpleToJava {
+    state_machine_StateDependentMethods_All <<!<</*state_machine_StateDependentMethods_All*/>><<#
+    // Issue #1351
+    if (!uClass.getStateDependentMethods().isEmpty()) {
+
+      // Effectively a lambda function below, so that we can keep code related
+      // to state-dependent method generation all in one space
+      abstract class StateDependentTreeTraversal {
+        public abstract void traverse(Method method);
+      }
+    
+      StateDependentTreeTraversal t = new StateDependentTreeTraversal() {
+    		
+        // Prints the signature for the state-dependent method
+        private void printBefore(Method method) {
+          String methodModifier = method.getModifier().equals("") ? "public" : method.getModifier();
+          String methodType = method.getType();
+          String methodName = method.getName();
+          StringBuilder parameters = new StringBuilder();
+          String finalParams = "";
+          if (method.hasMethodParameters())
+          {
+            for (MethodParameter aMethodParam : method.getMethodParameters()) 
+            {
+              String paramName = aMethodParam.getName();
+              String paramType = aMethodParam.getType();
+              String isList = aMethodParam.getIsList() ? " [] " : " ";
+              String aSingleParameter = paramType + isList + paramName;
+              parameters.append(aSingleParameter + ", ");
+            }
+        
+            finalParams = parameters.toString().substring(0, parameters.toString().length()-2);
+          }
+      
+          StringBuilder methodExceptionsBuilder = new StringBuilder();
+          if(method.getExceptions()!=null&&method.numberOfExceptions()>0)
+          {
+            methodExceptionsBuilder.append(" throws ");
+            String exceptioncomma = "";
+            for(String methodException:method.getExceptions())
+            {
+              if(!"".equals(methodException))
+              {
+                methodExceptionsBuilder.append(exceptioncomma);
+                methodExceptionsBuilder.append(methodException);
+                exceptioncomma = ",";
+              }
+            }
+          }
+
+          String methodExceptions = methodExceptionsBuilder.toString();
+          append(realSb, "\n  {0} {1} {2}({3}){4}", methodModifier, methodType, methodName, finalParams, methodExceptions);
+        }
+    
+        // Prints the state-dependent method as defined within the UmpleClass
+        public void traverse(Method method)
+        {
+          printBefore(method);
+          Map<State, MethodBody> stateMap = uClass.getStateDependentMethodBodyMap().get(method);
+          MethodBody defaultBody = stateMap.getOrDefault(null, null);
+          for (StateMachine sm : uClass.getStateMachines())
+          {
+            StringBuilder smBuilder = new StringBuilder();
+            append(smBuilder, "\n  {");

```

Comment:
This needs to be fixed, should be moved out of loop otherwise will fail with multiple sms